bedrock/bedrock-common/src/main/java/com/nukkitx/protocol/bedrock/handler/DefaultBatchHandler.java

```diff
@@ -25,6 +26,7 @@ public void handle(BedrockSession session, ByteBuf compressed, Collection<Bedroc
             BedrockPacketHandler handler = session.getPacketHandler();
             if (handler == null || !packet.handle(handler)) {
                 log.debug("Unhandled packet for {}: {}", session.getAddress(), packet);
+                ReferenceCountUtil.release(packet);
```

Comment:
Maybe a `try-finally` around the handler as well would be appropriate since we don't want the buffer to leak if the handling fails.

Reply:
Done.